### PR TITLE
Add index pages

### PR DESCRIPTION
**Pull request by jonludlam**

### CODE REVIEW

File: src/lib/pages.ml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    let* build_job = Current.Job.start_with ~pool:build_pool ~level:Mostly_harmless job in
+    Current.Job.log job "Using cache hint %S" cache_hint;
+    Capnp_rpc_lwt.Capability.with_ref build_job @@ fun build_job ->
+    let** _ = Current_ocluster.Connection.run_job ~job build_job in
+    let extract_hashes (git_hashes, failed) line =
+      match Git_store.parse_branch_info ~prefix:"HASHES" line with
+      | Some value -> (value :: git_hashes, failed)
+      | None -> (
+          match String.split_on_char ':' line with
+          | [ prev; branch ] when Astring.String.is_suffix ~affix:"FAILED" prev ->
```
Comment (TheLortex) [inline src/lib/pages.ml:78]: Why do you need to parse lines that contains `FAILED` ? This was used in prep to identify packages that failed to install, but in this situation it shouldn't be needed

File: src/pipelines/docs.ml [hunk 2]
```diff
@@ -238,7 +240,7 @@ let v ~config ~api ~opam () =
                |> Current.state ~hidden:true)
         |> Current.list_seq
         |> Current.map (List.filter_map Result.to_option)
-      and+ metadata = metadata in
+      and+ metadata = pages in
```
Comment (TheLortex) [inline src/pipelines/docs.ml:243]: Is `metadata` replaced by `pages` ? In that case I believe the unused variable / module should be removed

**Review by TheLortex — COMMENTED** [review]
Looks good, I have some questions:

File: src/lib/pages.ml (hunk 1 above)
Comment (jonludlam) [reply src/lib/pages.ml:78]: This was cut'n'pasted from prep, I think - the intention was to dedup and move it somewhere shared.

File: src/pipelines/docs.ml (hunk 2 above)
Comment (jonludlam) [reply src/pipelines/docs.ml:243]: Ah yes, metadata should be used by pages, I need to fix that.

File: src/lib/pages.ml (hunk 1 above)
Comment (TheLortex) [reply src/lib/pages.ml:78]: Ok ! In Misc there is `fold_logs` that can be extended for this use case which is basically identifying and grouping by prefixes

**Comment by TheLortex** [comment]
Included in https://github.com/ocurrent/ocaml-docs-ci/pull/36

**Pull request closed without merging**